rpi/Animal/lib/Animal.pm

```diff
@@ -29,107 +31,103 @@ Perhaps a little code snippet.
     my $foo = Animal->new();
     ...
 
-=head1 EXPORT
-
-A list of functions that can be exported.  You can delete this section
-if you don't export anything, such as for a purely object-oriented module.
-
 =head1 SUBROUTINES/METHODS
 
-=head2 function1
+=head2 named
 
-=cut
+Create an animal with a name
 
-sub speaks_english {
-  return 0;
-}
+=cut
 
-=head1 AUTHOR
+sub named {
+  my ( $class, $name ) = @_;
 
-dickyince, C<< <minnimince at googlemail.com> >>
+  my $self = ( name => $name, colour => _default_colour() );
 
-=head1 BUGS
+  bless $self, $class;
+}
 
-Please report any bugs or feature requests to C<bug-animal at rt.cpan.org>, or through
-the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Animal>.  I will be notified, and then you'll
-automatically be notified of progress on your bug as I make changes.
+=head2 _default_colour
 
+Returns the default colour of an animal
 
+=cut
 
+sub _default_colour {
```

Comment:
+Perl private method semantics. That said, `default_colour` might be useful for a caller, and doesn't provide access to mutable state you wouldn't want to expose.